### ISSUE TITLE
Unable to use IPv6 as BIG-IP mgmt IP in deployment manifest

### ISSUE THREAD

**Issue by tkam8** [F5Networks/k8s-bigip-ctlr#1216]
#### Description
CIS unable to connect to an IPv6 IP address that is configured in the controller deployment manifest file. 

#### Kubernetes Version
[root@k8s-master ~(kube-system kubernetes-admin)]# kubectl version --short
Client Version: v1.15.0
Server Version: v1.15.0

#### Controller Version
CIS 1.14

#### AS3 Version
AS3 3.18

#### BIG-IP Version
BIG-IP 14.1.0.2

#### Diagnostic Information

internal SR: C3230136

The config in question is in the CIS deployment manifest file in the args block:
```
args: [
            "--bigip-username=$(BIGIP_USERNAME)",
            "--bigip-password=$(BIGIP_PASSWORD)",
            "--bigip-url=https://[fddb:ab11:cd22:a101:0:0:0:10]",
```
Multiple formats were attempted below:

- without brackets and still got error. Here are the logs:

```
----snip----
2020/03/24 12:57:43 [DEBUG] [AS3] posting request to https://fddb:ab11:cd22:a101::10/mgmt/shared/appsvcs/declare/
2020/03/24 12:57:43 [ERROR] [AS3] REST call error: Post https://fddb:ab11:cd22:a101::10/mgmt/shared/appsvcs/declare/: invalid URL port "ab11:cd22:a101::10"
2020/03/24 12:57:44 [ERROR] [2020-03-24 12:57:44,145 __main__ ERROR] invalid literal for int() with base 10: 'ab11:cd22:a101::10'
----snip----
```

- Tried expanding the ipv6 addr but no good:

```
----snip----
2020/03/24 13:02:00 [DEBUG] [AS3] posting request to https://fddb:ab11:cd22:a101:0:0:0:10/mgmt/shared/appsvcs/declare/
2020/03/24 13:02:00 [ERROR] [AS3] REST call error: Post https://fddb:ab11:cd22:a101:0:0:0:10/mgmt/shared/appsvcs/declare/: invalid URL port "ab11:cd22:a101:0:0:0:10"
2020/03/24 13:02:00 [ERROR] [2020-03-24 13:02:00,958 __main__ ERROR] invalid literal for int() with base 10: 'ab11:cd22:a101:0:0:0:10'
----snip----
```

- It looks like it's using the colon as a delimiter for the ip and port so tried below but still NG.

```
----snip----
2020/03/24 13:07:14 [DEBUG] [AS3] posting request to https://[fddb:ab11:cd22:a101:0:0:0:10]:443/mgmt/shared/appsvcs/declare/
2020/03/24 13:07:14 [ERROR] [AS3] REST call error: Post https://[fddb:ab11:cd22:a101:0:0:0:10]:443/mgmt/shared/appsvcs/declare/: dial tcp [fddb:ab11:cd22:a101::10]:443: connect: cannot assign requested address
----snip----
```

Ping works:
```
[root@k8s-master manifests(kube-system kubernetes-admin)]# ping6 fddb:ab11:cd22:a101:0:0:0:10
PING fddb:ab11:cd22:a101:0:0:0:10(fddb:ab11:cd22:a101::10) 56 data bytes
64 bytes from fddb:ab11:cd22:a101::10: icmp_seq=1 ttl=64 time=0.294 ms
64 bytes from fddb:ab11:cd22:a101::10: icmp_seq=2 ttl=64 time=0.315 ms
64 bytes from fddb:ab11:cd22:a101::10: icmp_seq=3 ttl=64 time=0.266 ms
```

#### Workaround
use hostAliases to map a hostname to the ipv6 addr and then use the hostname as the arg

**Comment by velascoblasco** [comment]
CIS handles the IP address to string so it does not matter what we define as URL in in this variable we will use it for connecting to. The error shows response from destination:

2020/03/24 12:57:43 [ERROR] [AS3] REST call error: Post https://fddb:ab11:cd22:a101::10/mgmt/shared/appsvcs/declare/: invalid URL port "ab11:cd22:a101::10"

func (postMgr *PostManager) httpPOST(request *http.Request) (*http.Response, map[string]interface{}) {
    	httpResp, err := postMgr.httpClient.Do(request)
    	if err != nil {
    		log.Errorf("[AS3] REST call error: %v ", err)
    		return nil, nil
    	}
    	defer httpResp.Body.Close()

It seems http response from endpoint is assuming an IPv4 <IP>:<port> and hence IP fddb:ab11:cd22:a101::10 is parsed as <fddb>:<ab11:cd22:a101::10>

**Comment by mdditt2000** [comment]
Internal Jira CONTCNTR-1829

**Comment by trinaths** [comment]
Addressed with https://github.com/F5Networks/k8s-bigip-ctlr/pull/1996 

closing this issue. Fix available with CIS 2.6
